packages/transformers/js/core/src/collect.rs

```diff
@@ -689,7 +692,7 @@ impl Visit for Collect {
         return;
       }
       Expr::This(_this) => {
-        if self.in_module_this {
+        if !self.is_module && self.in_module_this {
           handle_export!();
         } else if !self.in_class {
```

Comment:
now this branch will run if `self.in_module_this && self.is_module` as well, tracking `this_exprs` in the top-level unnecessarily.

```suggestion
        if self.in_module_this {
          if !self.is_module { 
            handle_export!();
          }
        } else if !self.in_class {
```

Reply:
🤦 This is why I hate `else if` lol